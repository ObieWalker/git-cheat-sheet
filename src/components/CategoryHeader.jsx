import React from 'react';

const CategoryHeader = ({categoryTitle}) => {
  return (
      <div>
        <h4 className="white-text">{categoryTitle}</h4>
      </div>
    );
  }

export default CategoryHeader;
