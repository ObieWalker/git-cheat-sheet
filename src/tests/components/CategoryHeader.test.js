import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import CategoryHeader from '../../components/CategoryHeader';

let mounted;
let props;
let wrapper;
const categoryTitle = "Title";
const category = {

}

const getComponent = () => {
  if (!mounted) {
    props = {
      categoryTitle
    };
    mounted = shallow(<CategoryHeader {...props} />);
  }
  return mounted;
};

describe('categoryHeader Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

});

