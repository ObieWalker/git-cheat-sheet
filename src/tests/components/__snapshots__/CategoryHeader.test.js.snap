// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categoryHeader Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryHeader
    categoryTitle="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h4
        className="white-text"
      >
        Title
      </h4>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Title",
        "className": "white-text",
      },
      "ref": null,
      "rendered": "Title",
      "type": "h4",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4
          className="white-text"
        >
          Title
        </h4>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
          "className": "white-text",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h4",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
